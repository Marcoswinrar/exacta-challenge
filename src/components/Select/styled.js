import styled from 'styled-components'
import media from 'styled-media-query'
import { UserFormFieldStyle } from '../../config/styles'

export const SelectWrapper = styled.div`
  display: flex;
  flex-direction: column;
`
export const Select = styled.select`
  ${UserFormFieldStyle}
  width: 336px;
  border: 2px solid black;
  height: 36px;

  :invalid {
    border-color: red;
  }

  ${media.lessThan("320px")`
    width: 100%;
  `}
`
export const Label = styled.label`
  font-size: .8rem;
  text-transform: uppercase;
  font-weight: bolder;
`

export const Option = styled.option``