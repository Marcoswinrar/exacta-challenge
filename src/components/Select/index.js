import * as S from './styled'

const Select = ({ data, onChange, label }) => (
  <S.SelectWrapper>
    <S.Label htmlFor={label}>{label}</S.Label>
    <S.Select
      id={label}
      onChange={onChange}
    >
      {data && data.map((d, index) => (
        <S.Option
          key={index}
          value={d.value}>
          {d.label}
        </S.Option>
      ))}
    </S.Select>
  </S.SelectWrapper>
)

export default Select