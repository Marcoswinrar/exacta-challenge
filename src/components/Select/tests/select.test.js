import { render, screen, fireEvent } from "@testing-library/react"
import { act } from "react-dom/test-utils"
import Select from ".."

describe("Componente <Select/>", () => {

  it("Deve renderizar o componente sem problemas", () => {
    const { container } = render(<Select label={"Select"} />)

    expect(screen.queryByText(/Select/))
    expect(container).toMatchSnapshot()
  })

  it("Deve chamar onChange e selecionar valor correto ao clicar", () => {
    const onChange = jest.fn()

    const data = [
      { label: "Teste 01", value: "01" },
      { label: "Teste 02", value: "02" }
    ]

    render(
      <Select
        label={"Select"}
        data={data}
        onChange={onChange}
      />
    )

    const select = screen.getByLabelText(/Select/)

    act(() => fireEvent.change(select, { target: { value: "2" } }))

    expect(onChange).toBeCalledTimes(1)
    expect(screen.queryByText(/Teste 02/)).toBeInTheDocument()
  })
})