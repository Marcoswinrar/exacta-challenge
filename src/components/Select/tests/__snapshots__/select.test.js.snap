// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente <Select/> Deve renderizar o componente sem problemas 1`] = `
<div>
  <div
    class="sc-bczRLJ goEvZo"
  >
    <label
      class="sc-dkzDqf hNEPNr"
      for="Select"
    >
      Select
    </label>
    <select
      class="sc-gsnTZi eXqMhp"
      id="Select"
    />
  </div>
</div>
`;
